### PR TITLE
Color corrector improved

### PR DESCRIPTION
**Pull request by c3r1c3**
Updated branch for review. Please be critical.

### CODE REVIEW

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

File: plugins/obs-filters/data/color_correction_filter.effect [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+
+sampler_state textureSampler {
+	Filter   = Linear;
+	AddressU = Clamp;
+	AddressV = Clamp;
+};
+
+struct VertData
+{
```
Comment (RytoEX) [inline plugins/obs-filters/data/color_correction_filter.effect]: Style:  Opening curly brace ( { ) should go on the same line as the struct definition.  See also other similar lines below.

**Comment by RytoEX** [comment]
The commit titles should be formatted like this:
"\<component\>: [Fix/Add/Remove/\<verb\>] \<issue here\>"

In this case, the component is "obs-filters".

**Comment by c3r1c3** [comment]
Thank you RytoEx. Code updated with the additional formatting. Initial Git commit message updated as well.

File: plugins/obs-filters/color-correction-filter.c [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+	color_correction_filter_update(filter, settings);
+	return filter;
+}
+
+static void color_correction_filter_render(void *data, gs_effect_t *effect)
+{
+	struct color_correction_filter_data *filter = data;
+
+	if (!obs_source_process_filter_begin(filter->context, GS_RGBA,
+		OBS_ALLOW_DIRECT_RENDERING))
```
Comment (RytoEX) [inline plugins/obs-filters/color-correction-filter.c]: If you check plugins/obs-filters/color-filter.c for its [equivalent line](https://github.com/jp9000/obs-studio/blob/master/plugins/obs-filters/color-filter.c#L123) for this, it uses three tabs relative to its parent line.  I'm honestly not sure if two or three extra tabs is the desired style for this.

**Comment by RytoEX** [comment]
All of the commits are for files in plugins/obs-filters, so they should all be prefixed with "obs-filters" for the component part of the commit message title, assuming they remain as individual commits and are not squashed.

File: plugins/obs-filters/color-correction-filter.c (hunk 2 above)
Comment (c3r1c3) [inline plugins/obs-filters/color-correction-filter.c]: No, you're correct. It should be 3 tabs.

Fixed.

**Comment by c3r1c3** [comment]
I have updated the branch with the formatting changes suggested by RytoEx.

HomeWorld has reviewed it and made some suggestions on optimizations, which I am currently implementing.

**Comment by c3r1c3** [comment]
Did all the optimizations I can. Please review.

**Comment by c3r1c3** [comment]
Added a swath of comments, and reduced/removed a ton of variables/math/calls that I didn't need.

**Comment by RytoEX** [comment]
Are there any unexpected behaviors that would be caused on the user side from replacing a filter as done in d4cd14b?  Will existing filters still work exactly as they had before?  Can this new version still replicate the behavior of the old one, if desired?  What was the deciding factor in actually replacing the filter rather than supplementing it?

I ask these questions both for posterity and in preparation for questions arising on the support areas of the forums about why their colors on filtered sources look different or don't work anymore, if that would indeed happen.  I'm not questioning the validity of any of the decisions made, I'm just asking for clarification and understanding.  Sorry for the trouble.

**Comment by jp9000** [comment]
The only thing that may not be quite identical is contrast -- the brightness/gamma settings however are identical.

**Comment by c3r1c3** [comment]
It is missing the color tint/overlay option that was in the previous model. (Which is why I was surprised that the other one was yanked). I can quickly add that (and fix some grammatical errors in the comments) if you want.

As to the brightness and Gamma settings being the same, that's because the gamma code is directly from Jim, and while completely different the brightness code end up being mathematically the same.

The contrast is a touch different because of the weighting system/math.

**Comment by c3r1c3** [comment]
Created a new Branch with the missing feature: https://github.com/c3r1c3/obs-studio/blob/Color_Overlay/plugins/obs-filters/color-correction-filter.c

**Pull request closed without merging**